### ISSUE TITLE
Provide IOpenApiSchema interface

### ISSUE THREAD

**Issue by petriashev** [microsoft/OpenAPI.NET#542]
I am a supporter of MicroElements.Swashbuckle.FluentValidation project and I want to support NJsonSchema (NSwag).

Issue: https://github.com/micro-elements/MicroElements.Swashbuckle.FluentValidation/issues/63
One of the possible solutions is to adapt NJsonSchema to OpenApiSchema but it is a concrete type.

Can we extract IOpenApiSchema interface?

**Comment by darrelmiller** [comment]
We will probably have to make some changes to the OpenApiSchema model as we implement OpenAPI 3.1 support and support for JSON Schema 2020-12 draft.  We can consider an interface then.

Wouldn't it just be easier to create a factory that creates an OpenAPISchema from an NJsonSchema parameter?

**Comment by petriashev** [comment]
> Wouldn't it just be easier to create a factory that creates an OpenAPISchema from an NJsonSchema parameter?

Yes it is another option. This case I need convert NJsonSchema -> OpenAPISchema and back. 
At this moment I think more easy is to create some schema adapter with functionality I need. But I don't want to create Yet Another Interface That Common to other existing :)

Third option is to support two branches for OpenApi and NJsonSchema separately.

**Comment by baywet** [comment]
This was actually added in v2 due to the work on references resolution!
https://github.com/microsoft/OpenAPI.NET/blob/6f7e533712274815682adc10f39140004ec04c0c/src/Microsoft.OpenApi/Models/Interfaces/IOpenApiSchema.cs#L11

Let us know if you have any additional comments or questions.

Closing
